packages/lyra/src/lyra.ts

```diff
@@ -120,36 +160,53 @@ export class Lyra<TSchema extends PropertiesSchema = PropertiesSchema> {
     let i = 0;
     let j = 0;
 
+    const documentIdsFromNonSearchable = this.getDocumentIDsFromWhere({
+      boolIndices,
+      numericIndices,
+    });
+
     for (const term of tokens) {
       for (const index of indices) {
-        const documentIDs = await this.getDocumentIDsFromSearch({
+        let documentIDs = await this.getDocumentIDsFromSearch({
           ...params,
-          index,
-          term,
-          exact,
+          index: index,
+          term: term,
+          exact: exact,
+          where: where,
         });
 
+        // To avoid duplicates we need a subtraction between added document
+        // And retrieved ones.
+        // We check if this is the first
+        // We use non-null assertion because setIntersection will return null only if both set are null
+        // We are sure that documentsIDs is not null, so we are safew
+        documentIDs = setIntersection(
```

Comment:
Can you benchmark this particular line? I remember re-initializing and changing the type of a variable, which affects the JIT performance.

Reply:
Ok I will try to do a benchmark with a constant variable, let's see the difference